File: leetcode-python-go/891. 子序列宽度之和/main.py
from typing import List

nums = [2,1,3]

def sumSubseqWidths(nums: List[int]) -> int:
    nums.sort()
    res = 0
    MOD = 10 ** 9 + 7
    x, y = nums[0], 2
    for j in range(1, len(nums)):
        res = (res + nums[j] * (y - 1) - x) % MOD
        x = (x * 2 + nums[j]) % MOD
        y = y * 2 % MOD
    return (res + MOD) % MOD


print(sumSubseqWidths(nums))

File: leetcode-python-go/891. 子序列宽度之和/main.go
package main

import (
	"fmt"
	"sort"
)

func sumSubseqWidths(nums []int) int {
	const mod int = 1e9 + 7
	sort.Ints(nums)
	res, x, y := 0, nums[0], 2
	for _, num := range nums[1:] {
		res = (res + num*(y-1) - x) % mod
		x = (x*2 + num) % mod
		y = y * 2 % mod
	}
	return (res + mod) % mod
}

func main() {
	nums := []int{2, 1, 3}
	results := sumSubseqWidths(nums)
	fmt.Println(results)
}
